src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -200,23 +205,33 @@ public void initDragAndDrop() {
             tabHeaderArea.setOnDragDropped(event -> {
                 tabbedPane.getTabs().remove(dndIndicator);
 
-                boolean success = false;
-
                 List<Path> bibFiles = DragAndDropHelper.getBibFiles(event.getDragboard());
-                if (!bibFiles.isEmpty()) {
-                    for (Path file : bibFiles) {
-                        ParserResult pr = OpenDatabase.loadDatabase(file.toString(), Globals.prefs.getImportFormatPreferences(), Globals.getFileUpdateMonitor());
-                        addParserResult(pr, true);
-                    }
-                    success = true;
-                }
-
+                boolean success = loadBibFiles(bibFiles);
                 event.setDropCompleted(success);
                 event.consume();
             });
         });
     }
 
+    // Loading data within an UI related class is really bad practice. Therefore this logic should be moved somewhere else
```

Comment:
You can take a look at the OpenDatabaseAction, especially openTheFile and openFiles which deals with the same problem by creating a BackgroundTask for the file loading and then adding them to the base panel

Reply:
Thanks for your review. I'll delegate the load process of the files to the OpenDatabaseAction class as you suggested.